container/factory.go

```diff
@@ -108,6 +110,30 @@ func (ms MetricSet) Add(mk MetricKind) {
 	ms[mk] = struct{}{}
 }
 
+func (ms MetricSet) String() string {
+	values := make([]string, 0, len(ms))
+	for metric := range ms {
+		values = append(values, string(metric))
+	}
+	sort.Strings(values)
+	return strings.Join(values, ",")
+}
+
+func (ms *MetricSet) Set(value string) error {
+	*ms = MetricSet{}
+	if value == "" {
+		return nil
+	}
+	for _, metric := range strings.Split(value, ",") {
+		if AllMetrics.Has(MetricKind(metric)) {
+			(*ms).Add(MetricKind(metric))
+		} else {
+			return fmt.Errorf("unsupported metric %q specified", metric)
+		}
+	}
+	return nil
+}
```

Comment:
Can you add a comment to `MetricSet` and state that the struct is not thread safe? And make sure that the struct is not accessed from multiple goroutines in cAdvisor?

Reply:
I added couple of commits clarifying & documenting MericSet thread-safety (removing unused Append() method, changing Add() method to private one and documenting Set() method usage).